lib/internal/streams/legacy.js

```diff
@@ -111,4 +112,24 @@ function prependListener(emitter, event, fn) {
     emitter._events[event] = [fn, emitter._events[event]];
 }
 
+let defaultHighwaterMark = null
+ObjectDefineProperty(Stream, 'defaultHighwaterMark', {
+  get () {
+    return defaultHighwaterMark ?? Buffer.poolSize;
```

Comment:
The point about this being set to `0` with `Buffer.poolSize = 0` still stands, I think?